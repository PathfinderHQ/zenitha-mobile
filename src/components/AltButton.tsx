import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Dimensions, StyleSheet, Text, Image } from 'react-native';
import Colors from '../constants/colors';
import { google } from '../constants';

const { height, width } = Dimensions.get('screen');

interface Props {
    title: string;
    onPress?: () => void;
}

const AltButton: FC<Props> = ({ title, onPress }) => {
    return (
        <TouchableOpacity onPress={onPress} style={styles.container}>
            <Image source={google.link} style={styles.image} />
            <Text style={styles.text}>{title}</Text>
        </TouchableOpacity>
    );
};

export default AltButton;

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: Colors.input_background,
        alignItems: 'center',
        width: width / 1.1,
        height: height / 15,
        justifyContent: 'center',
        alignSelf: 'center',
        padding: 18,
        borderRadius: 5,
        marginVertical: 10,
    },

    text: {
        color: Colors.main_text,
        fontSize: 16,
    },

    image: {
        height: 25,
        width: 25,
        marginRight: 10,
    },
});
