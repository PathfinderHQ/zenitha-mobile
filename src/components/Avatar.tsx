import React, { FC } from 'react';
import { StyleSheet } from 'react-native';
import { Avatar as BaseAvatar } from 'react-native-paper';

const Avatar: FC = () => {
    return <BaseAvatar.Text size={120} label='ZM' color='#fff' style={styles.container} />;
};

export default Avatar;

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#000',
    },
});
