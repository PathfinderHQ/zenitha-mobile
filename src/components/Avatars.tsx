import React, { FC, useState } from 'react';
import { View } from 'react-native';
import { Avatar } from '@rneui/themed';
import * as ImagePicker from 'expo-image-picker';

type AvatarComponentProps = {
    onPress?: () => void;
};

const Avatars: FC<AvatarComponentProps> = ({ onPress }) => {
    const [image, setImage] = useState<string | null>(null);
    const pickImage = async () => {
        const result = await ImagePicker.launchImageLibraryAsync({
            mediaTypes: ImagePicker.MediaTypeOptions.All,
            allowsEditing: true,
            aspect: [4, 3],
            quality: 1,
        });

        if (!result.canceled) {
            setImage(result.assets[0].uri);
        }
    };
    return (
        <View>
            <Avatar
                size={64}
                rounded
                source={{ uri: 'https://randomuser.me/api/portraits/women/57.jpg' }}
                title='Bj'
                containerStyle={{ backgroundColor: 'grey' }}
            >
                {/* add navigation here */}
                <Avatar.Accessory size={23} onPress={pickImage} />
            </Avatar>
        </View>
    );
};

export default Avatars;
