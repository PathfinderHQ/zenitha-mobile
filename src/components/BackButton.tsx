import React, { FC } from 'react';
import { View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import IconButtonComponent from './IconButton';
import { Navigation } from '../types';

type BackButtonProps = {
    size?: number;
};

const BackButton: FC<BackButtonProps> = ({ size = 20 }) => {
    const { goBack } = useNavigation<Navigation>();

    return (
        <View>
            <IconButtonComponent size={size} icon='keyboard-backspace' color='#252525' onPress={goBack} />
        </View>
    );
};

export default BackButton;
