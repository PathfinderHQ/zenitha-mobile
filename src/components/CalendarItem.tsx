import { FC } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import * as dateFns from 'date-fns';

interface CalendarItemProps {
    date: Date;
    onSelectDate: (value: string) => void;
    selected: string;
}

const CalendarItem: FC<CalendarItemProps> = ({ date, onSelectDate, selected }) => {
    /**
     * use moment to compare the date to today
     * if today, show 'Today'
     * if not today, show day of the week e.g 'Mon', 'Tue', 'Wed'
     */
    const day = dateFns.isEqual(date, new Date()) ? 'Today' : dateFns.format(date, 'E');
    // get the day number e.g 1, 2, 3, 4, 5, 6, 7
    const dayNumber = dateFns.format(date, 'dd');

    // get the full date e.g 2021-01-01 - we'll use this to compare the date to the selected date
    const fullDate = dateFns.format(date, 'yyyy-MM-dd');

    return (
        <TouchableOpacity
            onPress={() => onSelectDate(fullDate)}
            style={[styles.card, selected === fullDate && { backgroundColor: '#343A40' }]}
        >
            <Text style={[styles.big, selected === fullDate && { color: '#fff' }]}>{day}</Text>
            <View style={{ height: 10 }} />
            <Text style={[styles.medium, selected === fullDate && { color: '#fff', fontWeight: 'bold', fontSize: 24 }]}>
                {dayNumber}
            </Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    card: {
        borderRadius: 8,
        borderColor: '#ddd',
        padding: 3,
        alignItems: 'center',
        height: 70,
        width: 60,
    },
    big: {
        fontWeight: 'bold',
        fontSize: 18,
    },
    medium: {
        fontSize: 16,
    },
});

export default CalendarItem;
