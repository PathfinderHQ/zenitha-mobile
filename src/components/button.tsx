import React, { FC } from 'react';
import { TextInput, TouchableOpacity } from 'react-native-gesture-handler';
import { Dimensions, StyleSheet, View, Text } from 'react-native';
import Colors from './colors';


const { height, width } = Dimensions.get('screen');

interface Props {
    title: string;
    onPress: () => void;

}

const App: FC<Props> = (props) => {
    return (
        <TouchableOpacity style={styles.container}>
            <Text style={styles.text}>{props.title}</Text>
        </TouchableOpacity>
    )
}

export default App;

const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.button_background,
        alignItems: 'center',
        width: width / 1.1,
        justifyContent: 'center',
        alignSelf: 'center',
        padding: 10,
        borderRadius: 5,
        marginVertical: 10
    },

    text: {
        color: Colors.button_text,
    }
})