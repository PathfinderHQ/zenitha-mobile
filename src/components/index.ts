import Input from './hook-form/Input';
import FormProvider from './hook-form/FormProvider';
import Button from './Button';
import HorizontalDivider from './HorizontalDivider';
import Otp from './otp/Otp';

export { Input, Button, HorizontalDivider, Otp, FormProvider };
