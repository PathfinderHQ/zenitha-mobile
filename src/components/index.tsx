import Input from './input'
import Button from './button'
import Colors from './colors'

export { Input, Button, Colors }