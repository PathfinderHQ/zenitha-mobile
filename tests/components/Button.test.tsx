/* eslint-disable  @typescript-eslint/no-empty-function */
import * as React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../../src/components';

it(`renders correctly`, () => {
    const tree = renderer.create(<Button title='Test' onPress={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
});
