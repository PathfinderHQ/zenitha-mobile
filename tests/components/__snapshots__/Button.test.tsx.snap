// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#000000",
        "borderRadius": 5,
        "height": 88.93333333333334,
        "justifyContent": "center",
        "marginVertical": 20,
        "opacity": 1,
        "padding": 10,
        "width": 681.8181818181818,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F8F9FA",
          "fontSize": 18,
        }
      }
    >
      Test
    </Text>
  </View>
</RNGestureHandlerButton>
`;
